src/lib/supplier/supplier.model.js

```diff
@@ -1,14 +1,27 @@
 const mongoose = require('mongoose');
 
 const supplierSchema = new mongoose.Schema({
+supplierType: {
+type: String,
+enum:['dj', 'catering', 'photographer'],
+required:true
+},
+rank: {
+    type: Number,
+    required: true
```

Comment:
rank is required?
if there is a new supplier with no ranks already, cant create doc in the db?

for start with some value like 0 you can use `default: 0`